### PR TITLE
Add RedShift DATASHARE support

### PR DESCRIPTION
**Pull request by rpr-ableton**
### Brief summary of the change made

Features:
- `ALTER DATASHARE`
- `CREATE DATASHARE`
- `DESC DATASHARE`
- `DROP DATASHARE`
- `SHOW DATASHARES` 

### Are there any other side effects of this change that we should be aware of?
Nope!

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_redshift.py [hunk 1]
```diff
@@ -1120,6 +1120,161 @@ class CreateSchemaStatementSegment(BaseSegment):
     )
 
 
+@redshift_dialect.segment()
+class AltereDatashareStatementSegment(BaseSegment):
+    """A `ALTER DATASHARE` statement.
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_redshift.py:1125]: Nit

```suggestion
    """An `ALTER DATASHARE` statement.
```

File: src/sqlfluff/dialects/dialect_redshift.py [hunk 2]
```diff
@@ -1120,6 +1120,161 @@ class CreateSchemaStatementSegment(BaseSegment):
...
+        "DATASHARE",
+        Ref("ObjectReferenceSegment"),
+        Sequence(
+            Ref.keyword("SET", optional=True),
+            "PUBLICACCESSIBLE",
+            Ref("EqualsSegment", optional=True),
+            OneOf(
+                "TRUE",
+                "FALSE",
+            ),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_redshift.py:1212]: There's a segment for this btw:

```suggestion
            Ref("BooleanLiteralGrammar")
```

**Review by tunetheweb — CHANGES_REQUESTED** [review]
LGTM. A couple of small suggestions.

**Comment by rpr-ableton** [comment]
Thanks for the review! I've implemented your suggestions and regenerated the relevant yml test file :slightly_smiling_face:

**Review by tunetheweb — APPROVED** [review]
Looks great - thanks!

**Pull request merged**